sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -250,6 +250,12 @@ class Analyzer(catalog: Catalog,
         Project(
           projectList.flatMap {
             case s: Star => s.expand(child.output, resolver)
+            case Alias(UnresolvedFunction(name1, clds), name2) if containsStar(clds) =>
+              val newClds = clds.flatMap {
+                case s: Star => s.expand(child.output, resolver)
+                case o => o :: Nil
+              }
+              Alias(UnresolvedFunction(name1, newClds), name2)() :: Nil
```

Comment:
How about refactoring this branch to this:

```scala
            case a @ Alias(f @ UnresolvedFunction(_, args), _) if containsStar(args) =>
              val expandedArgs = args.flatMap {
                case s: Star => s.expand(child.output, resolver)
                case o => o :: Nil
              }
              a.copy(child = f.copy(children = expandedArgs))() :: Nil
```

(Also, please avoid using confusing variable names like `clds`, `name1`, or `name2`).